interfaces/compatibility/compatibility.go

```diff
@@ -250,3 +189,40 @@ func CheckCompatibility(compat1, compat2 CompatField) bool {
 	}
 	return true
 }
+
+// checkExpressionCompatibility check if the provided labels fulfill the
+// conditions expressed in the abstract syntz tree root.
+func checkExpressionCompatibility(node *Node, labels []CompatField) bool {
+	if (node.Left == nil && node.Right != nil) || node.Left != nil && node.Right == nil {
```

Comment:
feels like some of the check and accesses here could be methods on Node?

Reply:
Done